protocol/tx/entry.go

```diff
@@ -73,6 +97,10 @@ func writeForHash(w io.Writer, c interface{}) error {
 	case bc.AssetID:
 		_, err := w.Write(v[:])
 		return errors.Wrap(err, "writing bc.AssetID for hash")
+
+	case *EntryRef:
```

Comment:
Ooh that's interesting. I'm going to give that a try separately before proceeding with this PR.